src/library/scala/collection/immutable/List.scala

```diff
@@ -382,6 +381,18 @@ sealed abstract class List[+A]
     None
   }
 
+  override def corresponds[B](that: collection.Seq[B])(p: (A, B) => Boolean): Boolean = {
+    var i = this
+    var j = that
+    while (!(i.isEmpty || j.isEmpty)) {
+      if (!p(i.head, j.head))
+        return false
+      i = i.tail
+      j = j.tail
```

Comment:
This implementation should only be used when `that` is a `LinearSeq`. In other cases it can be much slower than using an `Iterator` for `that`.